test/memoryPoolAPI.cpp

```diff
@@ -21,6 +21,10 @@
 #include <unordered_map>
 #include <variant>
 
+#ifdef UMF_PROXY_LIB_ENABLED
+#include "proxy_lib_new_delete.h"
```

Comment:
Why are we only including this file for tests? Do we want to add new/removed support to the proxy library, or just add more tests? Perhaps it would be a good idea to separate this PR in the latter case

Reply:
This header should be included only in the code that is linked with the proxy library on Windows - see: https://github.com/microsoft/mimalloc/tree/8f7d1e9a41bb0182166aac6a8d4d8b00f60ed032?tab=readme-ov-file#override_on_windows

"_For best performance on Windows with C++, it is also recommended to also override the new/delete operations (by including [mimalloc-new-delete.h](https://github.com/microsoft/mimalloc/blob/8f7d1e9a41bb0182166aac6a8d4d8b00f60ed032/include/mimalloc-new-delete.h) a single(!) source file in your project)._"

Reply:
Documentation added.